### DISCUSSION TITLE
关于获取音乐url以及检查音乐可用性

### DISCUSSION THREAD

**Discussion by xiaomougui** [Binaryify/NeteaseCloudMusicApi#1487] [Q&A]
获取到的好多音乐无url，无版权，音乐可用性为false，包括一些流行音乐，比如薛之谦的绅士，请问这正常吗？

**Comment by Binaryify** [comment]
要提供相关信息，服务器，请求参数

**Reply by xiaomougui** [reply]
意思是我需要给你发过去，我写的请求，是这个意思吗？

**Reply by xiaomougui** [reply]
https://wangyiyun-one.vercel.app+‘/check/music’ 
请求参数是id=32192436
https://wangyiyun-one.vercel.app+‘/song/url’ 
请求参数是id=32192436
我没有服务器，用的是电脑，在vscode上用Ajax发送的请求
这个是绅士(薛之谦)的id，在网易云官网可以听(且id一模一样)有版权，
使用第一个url，返回的结果是false，暂时没有版权
使用第二个url，返回的结果是url：null
歌曲绅士的id是通过搜索获取的，也就是https://wangyiyun-one.vercel.app+‘/search’ ，请求参数为‘keyword=绅士’

**Reply by xiaomougui** [reply]
补充一下，第一个url返回的结果为404

**Reply by Binaryify** [reply]
vercel 要加个 realIp 参数，比如 https://wangyiyun-one.vercel.app/song/url?id=32192436&realIP=116.25.146.177

**Reply by xiaomougui** [reply]
我加了自己的IP，也试了试你的IP，还是不行，返回为null，realIp参数是IPv4地址吗？
![问111](https://user-images.githubusercontent.com/94113626/156682643-50a7a260-7ad7-4f1a-baa6-8ce83122f887.PNG)

**Reply by xiaomougui** [reply]
检查音乐是否可用(/check/music)还是返回404

**Reply by Binaryify** [reply]
<img width="1362" alt="image" src="https://user-images.githubusercontent.com/12221718/156685145-394787fe-1b36-4dd4-a7dd-174ab7ec54d0.png">
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/12221718/156685162-8456fd2f-df7b-4427-922d-48da075390e4.png">
不是很明白你的问题在哪，我用你的链接都没问题

**Reply by xiaomougui** [reply]
抱歉，写错了，realIP写成realIp了所以导致返回结果为空，改了之后好了，谢谢你，一直在回复我的问题，谢谢

**Reply by Binaryify** [reply]
嗯嗯
